tests/components/shopping_list/test_init.py

```diff
@@ -731,3 +733,46 @@ async def test_clear_completed_items_service(hass: HomeAssistant, sl_setup) -> N
     await hass.async_block_till_done()
     assert len(hass.data[DOMAIN].items) == 0
     assert len(events) == 1
+
+
+async def test_sort_list_service(hass: HomeAssistant, sl_setup) -> None:
+    """Test sort_all service."""
+
+    for name in ("zzz", "ddd", "aaa"):
+        await hass.services.async_call(
+            DOMAIN,
+            SERVICE_ADD_ITEM,
+            {ATTR_NAME: name},
+            blocking=True,
+        )
+    await hass.async_block_till_done()
+
+    # sort ascending
+    events = async_capture_events(hass, EVENT_SHOPPING_LIST_UPDATED)
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SORT,
+        {ATTR_REVERSE: False},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
```

Comment:
The service call is already blocking

```suggestion
```

Reply:
have also removed the unnecessary async_block_till_done() from all previous test cases